### ISSUE TITLE
Backblaze B2 Files not deleted

### ISSUE THREAD

**Issue by en3r0** [ncw/rclone#496]
When syncing, old files are not deleted. I know folders will stay, but they still have files in them.

> What is your rclone version (eg output from `rclone -V`) 1.28

> Which OS you are using and how many bits (eg Windows 7, 64 bit) Debian GNU/Linux 7.10 (wheezy)

> Which cloud storage system are you using? (eg Google Drive) BackBlaze B2

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)  /usr/sbin/rclone sync --transfers 20 /backupweekly/ BackBlazeB2:ThisBackupWeekly

> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)

Skipping the first part which is full of identical files...
2016/05/23 15:18:26 18-05-2016/website.com.tar.gz: Sizes identical
2016/05/23 15:18:26 18-05-2016/website.com.tar.gz: Unchanged skipping
2016/05/23 15:18:26 B2 bucket VBackup: Waiting for transfers to finish
2016/05/23 15:18:26 B2 bucket VBackup: Waiting for deletes to finish (during+after)

Transferred:            0 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:               651
Transferred:            0
Elapsed time:        1.8s

2016/05/23 15:18:26 Go routines at exit 7

 

Any ideas?

**Comment by ncw** [comment]
Can you send me the full log (via email if you like) and the full command line too. 

Thanks Nick

**Comment by en3r0** [comment]
I sent those things to your email just now.

**Comment by ncw** [comment]
Thanks for that.  I don't see anything out of order in there.

I think you are probably looking in the web interface to see the files.  Remember that b2 keeps the old versions of your files.  So if I do this

```
$ mkdir /tmp/b2-test
$ echo "hello world" > /tmp/b2-test/hello.txt
$ rclone sync /tmp/b2-test b2:bdelete-test
2016/05/24 13:52:00 Local file system at /tmp/b2-test: Building file list
2016/05/24 13:52:00 B2 bucket bdelete-test: Building file list
2016/05/24 13:52:01 Waiting for deletions to finish
2016/05/24 13:52:01 B2 bucket bdelete-test: Waiting for checks to finish
2016/05/24 13:52:01 B2 bucket bdelete-test: Waiting for transfers to finish
2016/05/24 13:52:02 B2 bucket bdelete-test: Waiting for deletes to finish (during+after)

Transferred:           12 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:                 0
Transferred:            1
Elapsed time:        2.6s

$ rclone ls b2:bdelete-test
       12 hello.txt

$ rm /tmp/b2-test/hello.txt 

$ rclone sync /tmp/b2-test b2:bdelete-test
2016/05/24 13:52:30 Local file system at /tmp/b2-test: Building file list
2016/05/24 13:52:30 B2 bucket bdelete-test: Building file list
2016/05/24 13:52:31 B2 bucket bdelete-test: Waiting for checks to finish
2016/05/24 13:52:31 B2 bucket bdelete-test: Waiting for transfers to finish
2016/05/24 13:52:31 Waiting for deletions to finish
2016/05/24 13:52:31 B2 bucket bdelete-test: Waiting for deletes to finish (during+after)

Transferred:            0 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:                 1
Transferred:            0
Elapsed time:        2.7s

$ rclone ls b2:bdelete-test

```

All the files have gone according to rclone as expected.

However if I look in the web interface I see this

![b2](https://cloud.githubusercontent.com/assets/536803/15504575/8bde0994-21b7-11e6-81bd-b99beacd4739.png)

And if I click on it I can see that `hello.txt` is made up of two files, firstly its original contents, and on top of that a delete marker which means that the file appears as deleted to rclone so doesn't appear in the `rclone ls`.

![b2-1](https://cloud.githubusercontent.com/assets/536803/15504616/bdf2ff3e-21b7-11e6-8ab8-ed2d9a87b89a.png)

Now the obvious question is how to delete those old versions if you don't want them.  There is an issue about this #462

So in summary, I think b2 is working as intended, storing all the old versions of your files for you.

**Comment by en3r0** [comment]
I can confirm, part of that was just not being clear on how the indicate deleted files. Thanks!
